### PR TITLE
refactor: simplify flow topology boundaries

### LINKED ISSUE #10744: Delete the keyless-leftover migration branch (D5): loud throw on fresh launch + existing record
## Context

Follow-up from #10727 (docs: define-out-of-existence audit — invented edges and self-inflicted races). The keyless-leftover migration branch (`runToolUseFlow.ts:482-526`, ~45 LoC, self-described "defensive fallback only") currently mishandles both residual reachers silently — a fresh launch colliding with an existing record adopts a foreign run's record.

## Details

- Fresh launch + existing record throws `PersistedFlowStateError` — loud, and more correct for both residual cases (crafted execute args, 48-bit id collision).
- Drop the flow's `migrateSharedState` import; the retrieval boundary keeps sole ownership.
- Carry the §5b ledger discipline.

## References

- PR: #10727
- Plan doc: `docs/proposals/2026-08-16-define-out-of-existence.md` §1b, §5 item 2
- Umbrella: #10742

### LINKED ISSUE #10752: Reorder the detach emission site to define away the #10693 topology guard (D15)
## Context

Follow-up from #10727 (docs: define-out-of-existence audit — invented edges and self-inflicted races). The #10693 topology guard exists because `detachActiveChildren` emits per-child null edges before the retained-children roster, creating a detach-before-refresh interleaving that the marker-refresh third loop then patches.

## Details

- Reorder the one emitter: silent detach → `child.activity` roster (retention stamps normally) → explicit null edges (`setParentStream` already preserves retained placements). The marker-refresh third loop becomes unreachable and deletes (~30 LoC + comment mass).
- Riders: `sessionStores.onChildrenDetached` (the second null-edge emitter, fires on stream deletion) is safe but must not move ahead of roster emission; the reorder changes public NDJSON fact order (same information) — note for headless-parity review; and the implementing PR must re-confirm that after the null edges the detaching parent emits no further retained-bearing `child.activity` tick.
- Lands with or after the #10693 follow-up; resolves that PR's WATCH.
- Carry the §5b ledger discipline.

## References

- PR: #10727
- Plan doc: `docs/proposals/2026-08-16-define-out-of-existence.md` §1d, §5 item 4
- Guard PR: #10693
- Umbrella: #10742

### PR DESCRIPTION
**Pull request by LionSR**
## Summary

- reject fresh tool-use launches that collide with persisted flow state, leaving legacy migration solely at resume retrieval
- emit the retained child roster before parent-null topology facts, then delete the unreachable incompatible-row marker repair
- update the durable boundary tests for collision rejection, retrieval-owned migration, and public event order

The NDJSON fact order changes for child detachment, but the information is unchanged: `child.activity` is emitted after silent detach and before `setParentStream` null edges. The stream-deletion callback still calls `detachActiveChildren` first and emits additional null edges only for inactive children; the detaching parent emits no retained-bearing roster afterward.

This should merge with or after #10801 because both branches touch the tool-use flow setup area.

Closes #10744
Closes #10752
Part of #10742

## Issue acceptance gates

- #10744: a fresh tool-use launch that encounters a persisted flow record now rejects it; compatibility migration is owned only by resume retrieval.
- #10752: `detachActiveChildren` silently detaches, emits its retained-child roster, then emits null parent edges. The TUI marker-repair loop is removed, and no later retained-bearing roster is emitted for that parent.

## Single source of truth

- `SessionResumeRetrieval` is the single compatibility/migration boundary for persisted tool-use flow state.
- `ExecutionRegistry.detachActiveChildren` is the single ordered producer for its detach roster and null-edge sequence; `projectChildRoster` consumes the durable roster without a compensating topology race.

## Duplication and abstraction budget

Removed the fallback migration branch and the TUI marker-refresh repair. No new abstraction, export, or pass-through layer was added.

## Net elements (R6)

5 files changed: 53 additions, 82 deletions (net -29 LoC). No exported symbols, classes, interfaces, or enums were added.

## Consumer counts (R8)

No emit path or export is deleted or re-routed; only the order of the two existing facts changes. In-repo direct fact consumers are: `child.activity` — 4 (`SessionFactApplier`, `sessionProgressSubscription`, `StreamSnapshotStore`, `TexraTranscriptRecorder`); `setParentStream` — 3 (`SessionFactApplier`, `sessionProgressSubscription`, `StreamSnapshotStore`). The CLI projection reached through `sessionProgressSubscription` is `projectChildRoster`; external NDJSON readers observe the documented order change.

## Host impact

Extension: session-fact application retains the same information.

Desktop: no host-specific behavior changed.

CLI: child detach facts have a deterministic roster-before-null-edge order; the TUI no longer needs a repair loop.

## Scheduled refactoring

None.

## Validation

- corepack pnpm run typecheck
- corepack pnpm run lint
- corepack pnpm run compile:fast
- corepack pnpm test (878 passed, 1 skipped; 10,728 tests passed, 5 skipped)
- focused resume, execution-registry, and TUI projection suites (303 tests passed)
- git diff --check origin/main...HEAD

### CODE REVIEW

File: packages/cli/src/chat/tui/state/childExecutions.ts [hunk 1]
```diff
@@ -288,10 +288,8 @@ function subagentEntryUnchanged(
  * child it never retains (or a direct unit-level projection). A late roster
  * from an incompatible (explicitly edged-elsewhere, or promoted) child
  * cannot resurrect active membership or overwrite a newer parent's summary
- * metadata — but a *retained* row still refreshes the historical
- * placement's marker when that placement names this parent (post-promotion
- * roster refreshes land here, e.g. "keep subagents running"), because
- * retained membership follows the shared roster, not current topology.
+ * metadata. Retained membership follows the shared roster only while its
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/state/childExecutions.ts]: Clarified in 1776a23c7c. The comment now separates compatibility-gated admission and finishedAt refresh from the later retained-roster cap eviction.

File: src/agent/implementations/flows/tooluse/runToolUseFlow.ts [hunk 2]
```diff
@@ -479,48 +478,7 @@ export async function runToolUseFlow<C = unknown>(
...
-        if (parsedShared.changed) {
-          logger.debug('Normalized persisted tool-use shared state');
-        }
-        flowRecord.shared = sharedData;
-        await kv.write(
-          flowKey(executionId),
-          stampFlowRecordSchemaVersion(flowRecord),
-        );
-      }
+      throw new PersistedFlowStateError(executionId, 'invalid-shared');
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/agent/implementations/flows/tooluse/runToolUseFlow.ts]: Fixed at the current head ea79f5208b. The fresh-launch collision now parses the present shared record solely to retain a Zod cause when it is corrupt; a valid leftover record keeps the same invalid-shared behavior without a fabricated cause.

File: src/agent/implementations/flows/tooluse/runToolUseFlow.ts [hunk 3]
```diff
@@ -479,48 +479,12 @@ export async function runToolUseFlow<C = unknown>(
         );
       }
     } else if (flowRecord) {
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/agent/implementations/flows/tooluse/runToolUseFlow.ts:481]: Added unexpected-record as a distinct persisted-flow reason. A valid record on a fresh launch now reports the missing resume handoff; an actually invalid record still reports invalid-shared with its Zod cause.

File: src/agent/runtime/executionRegistry.ts [hunk 4]
```diff
@@ -546,21 +547,21 @@ export class ExecutionRegistry {
...
-      });
+      detachedChildStreamIds.push(activation.childStreamId);
     }
     for (const handle of this.handles.values()) {
       if (!isChildExecution(handle, parentStreamId)) continue;
       this.approvals?.detachStreamFromParent(handle.childStreamId);
       handle.detach();
+      detachedChildStreamIds.push(handle.childStreamId);
+    }
+    this.emitChildActivity(parentStreamId);
```
[the remark this answers is not in the document]
Comment (LionSR) [reply src/agent/runtime/executionRegistry.ts:558]: Updated the ordering census section to document the current roster-before-null-edge promotion order.

File: packages/cli/src/chat/tui/state/childExecutions.ts [hunk 5]
```diff
@@ -288,10 +288,9 @@ function subagentEntryUnchanged(
  * child it never retains (or a direct unit-level projection). A late roster
  * from an incompatible (explicitly edged-elsewhere, or promoted) child
  * cannot resurrect active membership or overwrite a newer parent's summary
- * metadata — but a *retained* row still refreshes the historical
- * placement's marker when that placement names this parent (post-promotion
- * roster refreshes land here, e.g. "keep subagents running"), because
- * retained membership follows the shared roster, not current topology.
+ * metadata. Retained rows are admitted and their `finishedAt` marker refreshed
+ * from the shared roster only while their topology remains compatible; the
+ * retained-membership sync below still applies roster cap eviction.
```
[the remark this answers is not in the document]
Comment (LionSR) [reply packages/cli/src/chat/tui/state/childExecutions.ts:293]: Preserved replay tolerance without weakening topology precedence: an incompatible retained roster can now stamp only the existing old-parent placement's finishedAt marker, so later omission propagates cap eviction without overwriting summary or resurrecting the edge. Added the edge-before-retained-roster regression.
